ddtrace/tracer/util.go

```diff
@@ -10,9 +10,13 @@ import (
 	"strings"
 )
 
-// toFloat64 attempts to convert value into a float64. If it succeeds it returns
-// the value and true, otherwise 0 and false.
+// toFloat64 attempts to convert value into a float64. If the value is an integer
+// greater or equal to 2^53 or less than or equal to -2^53, it will not be converted
+// into a float64 to avoid losing precision. If it succeeds in converting, toFloat64
+// returns the value and true, otherwise 0 and false.
 func toFloat64(value interface{}) (f float64, ok bool) {
+	const max = (int64(1) << 53) - 1
```

Comment:
These should be defined at the global level and used in tests too.

Reply:
> used in tests too
While I totally agree with moving them at the global level, I kind of disagree to use them in tests. Tests should use final numbers or their own constants in my opinion.
The reason for that is that by mistake you change it to `(int64(1) << 53) - 10` then your tests would still pass but the product is broken.
In other words you do not want to test that something holds true for a constant in the code, but you want to test that this holds true for the number  `(int64(1) << 53) - 1`

Reply:
Ok, now that you've explained it like this, I see what you mean. We want to make sure that if someone changes the constants, the tests catch that. Right. That's much better. Please ignore the comments then.